### ISSUE TITLE
"Error running flusher" in logs with v1.7.25

### ISSUE THREAD

**Issue by tmm1** [getsentry/sentry-java#772]
```
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: Error running Flusher: 
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: java.lang.NullPointerException
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.Objects.requireNonNull(Objects.java:203)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.Arrays$ArrayList.<init>(Arrays.java:3826)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.Arrays.asList(Arrays.java:3813)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at d.a.g.b.a(DiskBuffer.java:26)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at io.sentry.connection.c$c.run(BufferedConnection.java:3)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:278)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:273)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
10-02 08:58:32.128  8296  8359 E io.sentry.connection.c: 	at java.lang.Thread.run(Thread.java:761)
```

**Comment by bruno-garcia** [comment]
Could you please give some context on the environment you're using the SDK? Is this an Android app or spring or else?

From the stacktrace it seems `File.listFiles()` is returning `null`:

https://github.com/getsentry/sentry-java/blob/b5c38dae46f7dcf54c7477ec4910e7c2e854fb1a/sentry/src/main/java/io/sentry/buffer/DiskBuffer.java#L176

Looks like one of the reasons why it might return `null` is [lack of permissions to a directory](https://stackoverflow.com/questions/30805801/file-listfiles-returns-null).
Also, are you specifying the directory manually? The SDK will output the directory in the log:

https://github.com/getsentry/sentry-java/blob/dae5e1326a512f229488e382899ee63b204739b3/sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java#L193

**Comment by bruno-garcia** [comment]
We've just noticed that `DiskBuffer` actually verifies it has access to the directory:

https://github.com/getsentry/sentry-java/blob/b5c38dae46f7dcf54c7477ec4910e7c2e854fb1a/sentry/src/main/java/io/sentry/buffer/DiskBuffer.java#L42

So perhaps the directory was deleted or permissions changed at runtime?
More info on this would help us troubleshoot it.

Meanwhile we've opened [this pr to address the NPE](https://github.com/getsentry/sentry-java/pull/775).

**Comment by tmm1** [comment]
This is from an Android app. I am not providing any custom directory. Using a pretty basic Sentry.init as outlined in the Android guide.

**Comment by bruno-garcia** [comment]
Thanks for the details.
I'll close this as solved by #775
